### PR TITLE
Types: be more specific about required scalar types

### PR DESCRIPTION
**Pull request by Slamdunk**

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -3,23 +3,6 @@ parameters:
     paths:
         - src
         - tests
-    ignoreErrors:
```
Comment (Slamdunk) [inline phpstan.neon:6]: By letting `vendor/bin/phpstan analyse --generate-baseline` generating the errors, they are much more specific and so eventual future inconsistencies emerge earlier

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+		-
+			message: "#^Parameter \\#5 \\$query of class OTPHP\\\\Url constructor expects array\\<non\\-empty\\-string, mixed\\>, array\\<int\\|string, array\\|string\\> given\\.$#"
+			count: 1
+			path: src/Url.php
+
+		-
+			message: "#^Parameter \\#1 \\$counter of method OTPHP\\\\HOTP\\:\\:setCounter\\(\\) expects int\\<0, max\\>, \\-500 given\\.$#"
+			count: 1
+			path: tests/HOTPTest.php
```
Comment (Slamdunk) [inline phpstan-baseline.neon:71]: Some guard clause are now caught by static analisys, but better leave the code there and keep the tests with the Exception when real types mismatch the one required by `DocBlock`

File: phpstan.neon (hunk 1 above)
Comment (Spomky) [reply phpstan.neon:6]: Excellent! I was not aware of this command.
I tested it on another project and you are right, it helps a lot.

File: phpstan-baseline.neon (hunk 2 above)
Comment (Spomky) [reply phpstan-baseline.neon:71]: OK I agree with you.

**Comment by Spomky** [comment]
Many thanks for this PR!

**Comment by Slamdunk** [comment]
Hi, may I ask you to release this, maybe in the [`11.1.1`](https://github.com/Spomky-Labs/otphp/milestone/10) milestone?

**Comment by Spomky** [comment]
Hi,

I didn't pay attention, but this PR is based on 11.2.x and not 11.1.x.
So it will be 11.2.0. It's okay I guess, I won't mess up the branches.

**Pull request merged**